DEVPROD-5232: Add mention of timeout configurations specific to CLI tools running on Evergreen

PyGeek03 opened the pull request:
[DEVPROD-5232](https://jira.mongodb.org/browse/DEVPROD-5232)

### Description
Add mention of timeout configurations specific to CLI tools running on Evergreen (such as DSI), to help users who run performance tests avoid a common pitfall where they didn't know that they also have to set these DSI parameters for long-running tests.

ablack12 reviewed (changes requested):
@PyGeek03 based on the user's comment, it sounds like they want all three timeouts to handle perf changes documented in the same place. I'm wondering if a change was made or could be made instead to the perf files to indicate the Evergreen configurations.  Although this wording is okay, I'm a bit hesitant to have the ownership of the relationship here be on us.

PyGeek03 commented:
I've added [a mention of Evergreen's timeout configurations to the DSI wiki page](https://docs.devprod.prod.corp.mongodb.com/dsi/Configuration-Explained/#overridesyml:~:text=Note%3A%20Evergreen%20has%20its%20own%20timeout%20configurations%20(like%20exec_timeout_secs%20%26%20timeout_secs)%2C%20which%20might%20override%20DSI%27s%20timeout%20configurations%20documented%20on%20this%20page.%20These%20settings%20are%20documented%20here.). I just thought it might be a good idea to signal to users of CLI tools running on Evergreen (like DSI) that they should check those tools' documentation for timeout configurations specific to those tools. I've removed the link to the DSI wiki page, what do you think?

ablack12 reviewed (approved):
Reducing the link reduces the dependency enough for me. Thanks!

PyGeek03 commented:
Yes this is ready to be merged, thank you!
